src/main/scala/model/world/DisturbingFactor.scala

```diff
@@ -0,0 +1,170 @@
+package model.world
+
+import engine.SimulationConstants._
+import model.Bunny
+import model.genome.Alleles.AlleleKind
+import model.genome.Genes
+import model.genome.Genes.GeneKind
+import model.world.FactorsUtils.{filterBunniesWithAlleles, killPercentageBunnies}
+import model.world.Generation.Population
+import scala.util.Random._
+
+sealed trait Factor {
+
+  /** @return the normal damage that is applied with this Factor */
+  val normalDamage: Double
+
+  /** Method that applies the damage of this Factor on a Population with the given climate
+   * @param bunnies the actual Population
+   * @param climate the actual Climate
+   * @return the remaining Population */
+  def applyDamage(bunnies: Population, climate: Climate): Population
+
+}
+
+object Factor {
+
+  abstract class BasicFactor extends Factor {
+    override def applyDamage(bunnies: Population, climate: Climate): Population = killPercentageBunnies(bunnies, normalDamage)
+  }
+
+  sealed trait FoodFactor extends Factor {
+    override val normalDamage: Double = FOOD_FACTOR_NORMAL_DAMAGE
+    private val lowDamage: Double = FOOD_FACTOR_LOW_DAMAGE
+
+    /** The LimitedFood Factor is introduced into the environment
+     * @return true if LimitedFood is active */
+    var isLimitedFood: Boolean
+
+    /** The HighFood Factor is introduced into the environment
+     * @return true if HighFood is active */
+    var isHighFood: Boolean
+
+    /** @return The GeneKind afflicted by HighFood */
+    val highFoodDamagedGene: GeneKind = Genes.JUMP
+
+    /** The ToughFood Factor is introduced into the environment
+     * @return true if ToughFood is active */
+    var isToughFood: Boolean
+
+    /** @return The GeneKind afflicted by ToughFood */
+    val toughFoodDamagedGene: GeneKind = Genes.TEETH
+
+    abstract override def applyDamage(bunnies: Population, climate: Climate): Population = {
+      val bunniesSplitByHighFoodGene = Bunny.splitBunniesByGene(highFoodDamagedGene, bunnies)
+      val bunniesSplitByToughFoodGene = Bunny.splitBunniesByGene(toughFoodDamagedGene, bunnies)
+
+      (isHighFood, isLimitedFood, isToughFood) match {
```

Comment:
qui non capisco perchè in alcuni casi di fa il Kill vero e proprio mentre in altri si applica solo il damage

Reply:
perchè con `super.applyDamage` vado ad applicare l'implementazione base del metodo contenuta in `BasicFactor` che applica il danno "normale", mentre con il metodo `killPercentageBunnies` vado ad applicare una percentuale più bassa.

Reply:
Giusto! Forse potremmo uniformare questi due nomi per farlo capire? tipo `applyDamage` e `applyMoreDamage` (?) non convince nemmeno me il nome ma avere chiamarli in modo simile per me lo rende più leggibile

Reply:
che ne dici di lasciare quello base `applyDamage` e al posto di `killPercentageBunnies` chiamarlo `applyCustomDamage` ?